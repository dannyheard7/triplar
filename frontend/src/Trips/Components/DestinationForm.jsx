import React from "react";
import NonFieldErrors from "Forms/Components/NonFieldErrors";
import CitySearchFieldContainer from "Forms/Containers/CitySearchFieldContainer";
import DateFieldGroup from "Forms/Components/DateFieldGroup";


export default function DestinationForm(props) {
     const itinerary = (props.itinerary) ? props.itinerary : {};
     let location = (typeof itinerary.city !== 'undefined') ? itinerary.city.name_std + ", " + itinerary.city.country : "";

    return (
        <form onSubmit={props.onSubmit}>
            <NonFieldErrors errors={props.errors.non_field_errors}/>
            <CitySearchFieldContainer errors={props.errors.locations} value={location} name="city"/>
            <DateFieldGroup errors={props.errors.start_date} label="Arrival" name="start_date"
                            value={itinerary.start_date} required="true"/>
            <DateFieldGroup errors={props.errors.end_date} label="Departure" name="end_date"
                           value={itinerary.end_date} required="true"/>
            <button type="submit" className="btn btn-primary">{props.submitLabel}</button>
        </form>
    );
}
