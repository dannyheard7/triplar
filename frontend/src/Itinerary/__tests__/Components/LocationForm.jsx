import React from "react";

import DestinationForm from "Itinerary/Components/LocationForm";
import ShallowRenderer from "react-test-renderer/shallow";

const renderer = new ShallowRenderer();



describe('<DestinationForm />', () => {
    const props = {
        location: {
            city: {
                name_std: "Bristol",
                country: "United Kingdom"
            },
        },
        errors: {}
    };

    test('renders correctly', () => {
        const result = renderer.render(<DestinationForm {...props} />);
        expect(result).toMatchSnapshot();
    });
});