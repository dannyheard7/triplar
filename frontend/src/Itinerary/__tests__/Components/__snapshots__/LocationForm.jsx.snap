// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationForm /> renders correctly 1`] = `
<form>
  <NonFieldErrors />
  <CitySearchFieldContainer
    name="city"
    value="Bristol, United Kingdom"
  />
  <DateFieldGroup
    label="Arrival"
    name="startDate"
    onChange={[Function]}
    required={true}
    value="01/12/2018"
  />
  <DateFieldGroup
    label="Departure"
    name="endDate"
    onChange={[Function]}
    required={true}
    value="02/12/2018"
  />
  <button
    className="btn btn-primary"
    type="submit"
  />
</form>
`;
